compiler/runtime.ml

```diff
@@ -172,19 +172,46 @@ let embed_date x = Date x
 let embed_duration x = Duration x
 let embed_array f x = Array (Array.map f x)
 
-type event =
-  | BeginCall of string list
-  | EndCall of string list
-  | VariableDefinition of string list * runtime_value
+type information = string list
+
+type raw_event =
+  | BeginCall of information
+  | EndCall of information
+  | VariableDefinition of information * runtime_value
   | DecisionTaken of source_position
 
-let log_ref : event list ref = ref []
+type event =
+  | VarDef of var_def
+  | VarDefWithFunCalls of var_def_with_fun_calls
```

Comment:
I would be in favor of merging `VarDef` and `VarDefWithFunCall` into a single case `VarComputation`.

Reply:
Actually there aren't two kinds. The situation is clearer when you say "When computing variable definitions, there may be intermediate function calls part of the log".

Reply:
Okay, I will modify this :ok_hand: